Sources/PostgresNIO/New/PostgresRowSequence.swift

```diff
@@ -60,13 +64,48 @@ extension PostgresRowSequence {
 extension PostgresRowSequence.AsyncIterator: Sendable {}
 
 extension PostgresRowSequence {
+    /// Collect and return all rows.
+    /// - Returns: The rows.
     public func collect() async throws -> [PostgresRow] {
         var result = [PostgresRow]()
         for try await row in self {
             result.append(row)
         }
         return result
     }
+
+    /// Collect and return all rows, alongside the query metadata.
+    /// - Returns: The query metadata and the rows.
+    public func collectWithMetadata() async throws -> (metadata: PostgresQueryMetadata, rows: [PostgresRow]) {
+        let rows = try await self.collect()
+        guard let metadata = PostgresQueryMetadata(string: self.rowStream.commandTag) else {
+            throw PSQLError.invalidCommandTag(self.rowStream.commandTag)
+        }
+        return (metadata, rows)
+    }
+
+    /// Consumes all rows and returns the query metadata.
+    ///
+    /// If you don't need the query metadata, just use the for-try-await-loop syntax:
+    /// ```swift
+    /// for try await row in myPostgresSequence {
+    ///     /// Process each row
+    /// }
+    /// ```
+    ///
+    /// - Parameter onRow: Processes each row.
+    /// - Returns: The query metadata.
+    public func consume(
+        onRow: @Sendable (PostgresRow) throws -> ()
+    ) async throws -> PostgresQueryMetadata {
```

Comment:
I'm open to better names.

Reply:
Also the closure is intentionally not `async` since it's generally not a good idea to be doing `async` things on each row.
It's still possible to accumulate the rows and do whatever you want after.